examples/lib/stories/input/keyboard_listener_component_example.dart

```diff
@@ -2,131 +2,82 @@ import 'package:examples/commons/ember.dart';
 import 'package:flame/components.dart';
 import 'package:flame/game.dart';
 import 'package:flame/input.dart';
-import 'package:flame/palette.dart';
 import 'package:flutter/services.dart';
-import 'package:flutter/widgets.dart';
 
 class KeyboardListenerComponentExample extends FlameGame
     with HasKeyboardHandlerComponents {
   static const String description = '''
     Similar to the default Keyboard example, but shows a different
     implementation approach, which uses Flame's
     KeyboardListenerComponent to handle input.
-    Usage: Use A S D W to steer Ember.
+    Usage: Use WASD to steer Ember.
   ''';
 
-  static final Paint white = BasicPalette.white.paint();
-  static const int speed = 200;
+  static const int _speed = 200;
 
-  late final Ember ember;
-  final Vector2 velocity = Vector2(0, 0);
+  late final Ember _ember;
+  final Vector2 _direction = Vector2.zero();
+
+  final Map<LogicalKeyboardKey, double> _keyStrengths = {
```

Comment:
Will this ever need the double? Can't it just be a set of active keys instead?

Reply:
Right now these doubles are used in `xInput` and `yInput` getters. It can certainly be achieved with just a set of active keys as well. But then a lot of branching code will have to go inside `update` (something similar to what previous code was doing in _handleKey). For example, to detect horizontal movement, it will look like this,

if A and D are pressed => direction.x = 0
else if A is pressed but D is not pressed => direction.x = -1
else if D is pressed but A is not pressed => direction.x = 1

And same will have to be done for vertical movement as well. I think that unnecessarily makes the example look complex and lengthy

Reply:
Option 1: Rename to one of these:
- keyWeights
- keyAmounts
- keyValues 
- any other better suggestion

Option 2: I can convert the map to store Boolean instead of double and call it
- keyPressStates
- keyStates

The second option will make the `xInput` and `yInput` getters a little elaborate, but I'm still inclined towards doing that. What do you think?

Reply:
`keyWeights` I like the best I think. Naming is so god damn hard. :sweat_smile:

Reply:
Renamed to keyWeights